client/src/main/java/io/pravega/client/security/auth/JwtTokenProviderImpl.java

```diff
@@ -132,17 +134,22 @@ private JwtTokenProviderImpl(Controller controllerClient, String scopeName, Stri
     @Override
     public CompletableFuture<String> retrieveToken() {
         DelegationToken currentToken = this.delegationToken.get();
-
+        final CompletableFuture<String> result;
         if (currentToken == null) {
-            return this.refreshToken();
+            result = this.refreshToken();
         } else if (currentToken.getExpiryTime() == null) {
-            return CompletableFuture.completedFuture(currentToken.getValue());
+            result = CompletableFuture.completedFuture(currentToken.getValue());
```

Comment:
TokenExpirySignal has a higher priority and hence should be checked before returning a token right ?

Reply:
If the token has no expiry time set on it, the signal wouldn't be interpreted as token having been expired. Therefore, we do this check before the check for the signal.